src/Shared/Data/Database/Companions.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Linq;
+using Melia.Shared.Game.Const;
+using Newtonsoft.Json.Linq;
+using Yggdrasil.Data.JSON;
+
+namespace Melia.Shared.Data.Database
+{
+	public class CompanionData
+	{
+		public int Id { get; set; }
+		public string ClassName { get; set; }
+		public string Name { get; set; }
+		public float RideMSPD { get; set; }
+		public bool IsPremium { get; set; }
+		public bool CanPet { get; set; }
+		public bool CanRide { get; set; }
+		public bool EndRideOnHit { get; set; }
+		public int FoodGroup { get; set; }
+		public BuffId Buff { get; internal set; }
+		public int Price { get; set; }
+		public string FeedAnimation { get; set; }
+		public int FeedSleep { get; set; }
+		public bool IsRideOnly { get; set; }
+	}
+	public class CompanionDb : DatabaseJsonIndexed<int, CompanionData>
+	{
+		public bool TryFindByClassName(string className, out CompanionData data)
+		{
+			data = this.Entries.Values.FirstOrDefault(a => a.ClassName == className);
+			return data != null;
+		}
+
+		protected override void ReadEntry(JObject entry)
+		{
+			// Values: id, className, name, rideMSPD, premium, pet, ride, endRideOnHit, [foodGroup], [buff]
+			entry.AssertNotMissing("id", "className");
+
+			var data = new CompanionData();
+
+			data.Id = entry.ReadInt("id");
+			data.ClassName = entry.ReadString("className");
+			data.Name = entry.ReadString("name");
+			if (entry.ContainsKey("price"))
+				data.Price = entry.ReadInt("price");
```

Comment:
Is there a reason you're explicitly checking for the existence of keys instead of utilizing the default value overload of the read methods?